test/e2e/upgrade_test.go

```diff
@@ -32,6 +32,15 @@ func runSimpleUpgradeFlow(test *framework.ClusterE2ETest, updateVersion v1alpha1
 	test.DeleteCluster()
 }
 
+func runUpgradeFlowDiffCliVersionFromMain(test *framework.ClusterE2ETest, updateVersion v1alpha1.KubernetesVersion, clusterOpts ...framework.ClusterE2ETestOpt) {
```

Comment:
I'm ok with this for now, but shouldn't we make this dynamic so it works in release branches as well?
Like if the current branch is main, it uses `ExecuteWithLatestMinorReleaseFromMain`
But if it's a release branch, it uses something different

Reply:
Yeah, good point. I was originally going to have a separate function for the release branch tests called `runUpgradeFlowDiffCliVersionFromReleaseBranch`. And in that one, we would use the method that we already have in place called `ExecuteWithLatestMinorReleaseFromVersion`.

But making it dynamic seems to make more sense so when I add the tests for release branches in a separate PR, I will look into making this dynamic